### PR TITLE
feat: add file checking

### PR DESCRIPTION
**Pull request by benjaminlim00**

### CODE REVIEW

File: src/components/DropZone.tsx [hunk 1]
```diff
@@ -1,9 +1,27 @@
 import React from "react";
 import Dropzone from "react-dropzone";
 
+const onFileDrop = (acceptedFile: any): void => {
+  const reader = new FileReader();
+  if (reader.error) {
```
Comment (Nebulis) [inline src/components/DropZone.tsx:6]: in which case could this happen ? you created the reader just above and nothing happened in between ?

File: src/components/DropZone.tsx [hunk 2]
```diff
@@ -1,9 +1,27 @@
...
+    alert(`Error occurred: ${reader.error}`);
+  }
+  reader.onload = () => {
+    try {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+      //@ts-ignore
```
Comment (Nebulis) [inline src/components/DropZone.tsx:12]: why ?

Comment (benjaminlim00) [reply src/components/DropZone.tsx:12]: ts complains that the value of reader.result could be null

Comment (Nebulis) [reply src/components/DropZone.tsx]: Isn't this a valid complaint?

File: src/components/DropZone.tsx [hunk 3]
```diff
@@ -1,9 +1,32 @@
 import React from "react";
 import Dropzone from "react-dropzone";
 
+const onFileDrop = (files: File[]): void => {
+  const reader = new FileReader();
+  try {
```
Comment (Nebulis) [inline src/components/DropZone.tsx:6]: did you try ? it doesn twork when a file is not a json

File: src/components/DropZone.tsx [hunk 4]
```diff
@@ -1,9 +1,30 @@
...
+  reader.onerror = () => {
+    alert(`The file uploaded is not a valid TradeTrust file, error: ${reader.error}`);
+  };
+
+  reader.onload = () => {
+    try {
+      if (reader.result && typeof reader.result === "string") {
+        JSON.parse(reader.result);
+        alert(`The file ${files[0].name} will be processed`);
+      }
```
Comment (Nebulis) [inline src/components/DropZone.tsx:16]: else alert ?

Comment (benjaminlim00) [reply src/components/DropZone.tsx:18]: i have added it

Comment (Nebulis) [reply src/components/DropZone.tsx:18]: tradetrust ? what's that ? change to open attestation

Comment (benjaminlim00) [reply src/components/DropZone.tsx:18]: updated

File: .vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (Nebulis) [inline .vscode/settings.json:1]: to remove

**Pull request merged**